Sharpmake/Options.Vc.cs

```diff
@@ -452,9 +452,12 @@ public enum CppLanguageStandard
                     [Default]
                     CPP11,
                     CPP14,
+                    CPP17,
                     GNU98,
                     GNU11,
-                    GNU14
+                    GNU14,
```

Comment:
The change is good, but it's kind of weird that we had GNU for VC++ in the first place. The CPPXY use the same language standard as GNUXY on VC too so it seems useless. I think someone copy-pasted this from the Xcode compiler options. If I can confirm this (@belkiss), I think I will [Obsolete] them.

Reply:
Agreed, that file is called Option.Vc after all, implying that those values should be the correct ones from visual studio.

Reply:
Shall i remove the `GNU17` option if the others will be made `[Obsolete]`? Also is there a separate set of options for Clang, where `GNUXX` is meaningful?

Reply:
Hmm... I don't know if we have a generator that builds with Clang, or if there are plans to do that. There are a lot of advantages to using Clang even on Windows. In light of that, it may be worth keeping it for now and figure out later if we remove/deprecate them. I would just merge it as it is, but I will let belkiss decide.